torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -749,7 +793,7 @@ def _get_sfdp_patterns():
                 "replace_fn": replacement,
                 "example_inputs": args,
                 "trace_fn": joint_fwd_bwd,
-                "pass_dicts": patterns,
+                "pass_dicts": patterns[1],
```

Comment:
What is this?

Reply:
In order to share attention patterns among various implementations (viz. oneDNN Graph, ATen CPU SDPA, and ATen CUDA SDPA), I added another `PatternMatcherPass` in `joint_graph_passes.py`. It was necessary because in one `PatternMatcherPass`, it's not currently possible to have multiple replacements for the same pattern.
This approach is also being used in [`freezing_patterns.py`](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/freezing_patterns.py#L24-L29).


As for `joint_graph_passes`, the first `PatternMatcherPass` is (also) being used by oneDNN Graph to make any replacements if:
1. oneDNN Graph Inductor config is enabled & oneDNN Graph is supported, and
2. Pattern matches

The second `PatternMatcherPass` is now exclusively being used for the attention fusions in `fuse_attention.py`

I could add a comment at all the places where I had to change `patterns` to `patterns[1]`.